refactor

sshanks-kx opened the pull request:
further improvements to site to add clarity, fixes, etc

sshanks-kx committed:
further clarity to ipc messages

sshanks-kx committed:
more clarity around logging

sshanks-kx committed:
manual log file handling details

sshanks-kx committed:
improved -11! definition

sshanks-kx committed:
remove repeating info, add links to log

sshanks-kx committed:
fix link, remove dup info

sshanks-kx committed:
examples incorrect

sshanks-kx committed:
remove duplicate log replay info

sshanks-kx committed:
add direct links

sshanks-kx committed:
add converting bad to good log, change naming

sshanks-kx committed:
refactor tp log file details

sshanks-kx committed:
at end of doc, fix links no longer present

sshanks-kx committed:
removed caution text

sshanks-kx committed:
remove repeating info

sshanks-kx committed:
renamed kdb-tick to kdb+tick

sshanks-kx committed:
removed some txt

sshanks-kx committed:
hdb info

sshanks-kx committed:
rte examples

sshanks-kx committed:
updated with rte move example

sshanks-kx committed:
removed abbreviation

sshanks-kx committed:
fix broken links & improve txt

sshanks-kx committed:
add link to sys cmds on system keyword page + reformat

sshanks-kx committed:
add details of vwap examples

sshanks-kx committed:
add info on lvl2

sshanks-kx committed:
improved some info

sshanks-kx committed:
split c.q into sections so functionality noticed

sshanks-kx committed:
removed some unneeded txt

sshanks-kx committed:
clarity around dir to use

sshanks-kx committed:
improve rte info

sshanks-kx committed:
update rte example schema/feed so that it'll work with c.q

sshanks-kx committed:
moved rte examples to single page

natalietanner commented on docs/architecture/index.md:
```diff
@@ -137,6 +137,29 @@ The dates are referred to as _partitions_ and this on-disk structure contributes
+
+!!! Note "Standard kdb+ [command line options](../basics/cmdline.md) may also be passed"
+
+A HDB will be empty until the RDB saves its first set of tables at end-of-day.
```
```suggestion
A HDB is empty until the RDB saves its first set of tables at end-of-day.
```

natalietanner commented on docs/basics/ipc.md:
```diff
@@ -113,7 +113,24 @@ q)h"2+2" / this is sent to the remote process for calculation
+!!! tip "Default handling of a sync message"
+    [`.z.pg`](../ref/dotz.md#zpg-get) is called on the server when any message is received. 
+    The default processing is to call [`value`](../ref/value.md) with the provided message. Using the
+    documentation for `value`, the processing for the two messages above can be shown as:
```
```suggestion
    The default processing calls [`value`](../ref/value.md) with the provided message. Using `value`, the processing for the two messages above would be:
```

natalietanner commented on docs/basics/ipc.md:
````diff
@@ -185,7 +202,12 @@ Serializes and puts a message on the output queue for handle `h`, and does not b
 ```
 
-As per [sync messages](#sync-request-get), you can replace the string repesenting the code to execute as a list that forms a parse tree.
+As per [sync messages](#sync-request-get), you can replace the string repesenting the code to execute as a list. 
````
```suggestion
Similar to [sync messages](#sync-request-get), you can replace the string representing the code to execute as a list. 
```

natalietanner commented on docs/basics/syscmds.md:
```diff
@@ -41,8 +41,7 @@ Commands with optional parameters that set values, will show the current values
 
-!!! tip "The [`system`](../ref/system.md) keyword executes a string representation of a system command and returns its result."
-
+!!! tip "An alternative method to executing system commands is via the [`system`](../ref/system.md) keyword. This  executes a string representation of a system command and returns its result."
```
```suggestion
!!! tip "An alternative method to executing system commands is to use the [`system`](../ref/system.md) keyword. This executes a string representation of a system command and returns its result."
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -5,19 +5,29 @@ keywords: kdb+, log, logging, q, replication, recovery
 
+## Overview
+
+Software or hardware problems can cause a kdb+ server process to fail, possibly resulting in loss of data not saved to disk at the time of the failure. A kdb+ server can use logging of updates to avoid data loss when failures occur.
```
```suggestion
Software or hardware problems can cause a kdb+ server process to fail, possibly resulting in loss of data if not saved to disk at the time of the failure. A kdb+ server can use logging updates to avoid data loss when failures occur.
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -71,25 +77,21 @@ However, the checkpoint is path-dependent. Consider the following:
 
-Back in q:
+The system command `\l` can be used to roll the log file. 
+The current log file will be renamed with the `qdb` extension and a new log file will be created.
```
```suggestion
The current log file is renamed with the `qdb` extension and a new log file is created.
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -147,27 +149,27 @@ q logTest -l
+### Errors and rollbacks
 
-If either message handler (`.z.pg` or `.z.ps`) throws any error, and the state was changed during that message processing, this will cause a rollback.
+If either message handler ([`.z.pg`](../ref/dotz.md#zpg-get) or [`.z.ps`](../ref/dotz.md#zps-set)) throws any error, and the state was changed during that message processing, this will cause a rollback.
```
```suggestion
If either message handler ([`.z.pg`](../ref/dotz.md#zpg-get), or [`.z.ps`](../ref/dotz.md#zps-set)), throws any error and the state was changed during that message processing, this initiates a rollback.
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+This technique is often used to allow a developer more control over log file naming conventions, what to log, log file locations and the ability to add logic around the log file lifecycle.
+
+### Creating a log file
+
```
To create a log file, do the following.

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+
+### Creating a log file
+
+A log file can be initialized using [`set`](../ref/get.md#set).
```
```suggestion
1. Initialize a log file by using [`set`](../ref/get.md#set).
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+!!!note "`logfile set ();` is equivalent to `.[logfile;();:;()];`"
+
+A check for a pre-existing log file can be achieved using [`get`](../ref/get.md#get). 
+The script can now be written to initialize a new log file if it does not exist, otherwise it will opened for appending.
```
```suggestion
1. Check for pre-existing log files by using [`get`](../ref/get.md#get). 
1. If a new log file does not exist, the script can now be written to initialize it, otherwise it is opened for appending.
```

natalietanner commented on docs/kb/logging.md:
````diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+q)logfilehandle:hopen logfile;
+```
+
+Close the log file when finished logging any messages.
````
```suggestion
1. Close the log file when you are finished logging any messages.
```

sshanks-kx committed:
removed unneeded txt, rearranged nav

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+
+Function calls and the contents of the their parameters can be recorded to a log file, which can then be replayed by a process. This is often used for data recovery.
+
+This technique is often used to allow a developer more control over log file naming conventions, what to log, log file locations and the ability to add logic around the log file lifecycle.
```
```suggestion
This technique allows more control over actions like log file naming conventions, what to log, log file locations and the ability to add logic around the log file lifecycle.
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+
+This technique is often used to allow a developer more control over log file naming conventions, what to log, log file locations and the ability to add logic around the log file lifecycle.
+
+### Creating a log file
```
```suggestion
### Create a log file
```

natalietanner commented on docs/kb/logging.md:
````diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+q)hclose logfilehandle
+```
 
+### Log writting
````
```suggestion
### Log writing
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
 
+### Log writting
+
+To record events/messages to a log, append a list consisting of a function name followed by any parameters used. 
```
```suggestion
To record events and messages to a log, you must append a list consisting of a function name, followed by any parameters used. 
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+
+A tickerplant uses this concept to record all messages sent to its clients so they can use the log to recover. It records calling a function `upd` passing the parameters of a table name and the table content to append.
+
+For example, calling a function called `upd` with two parameters, x and y, can be recorded to a file file as follows:
```
```suggestion
For example, calling a function called `upd` with two parameters, x and y, can be recorded to a file as follows:
```

natalietanner commented on docs/kb/logging.md:
````diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+```q
+q)logfilehandle enlist (`upd;x;y)
+```
+When a kdb+ process plays back this log, a function called `upd` will be called with the value of the two parameters. Multiple function calls can be logged also:
````
```suggestion
When a kdb+ process plays back this log, a function called `upd` is called with the value of the two parameters. Multiple function calls can be logged also:
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+Function calls that are used to updating data are typically written, without logging the function definition. 
+This has the disadvantage of requiring that the recoverying process defines the functions (e.g. by loading a q script) prior to replaying the log. 
+The advantages can outweight the disadvantages by allowing for bugs fixes within the function or temporarily assigning the function to a different definition prior to playback, 
+to provide despoke logic for data sourced from a log file.
```
```suggestion
This has the disadvantage of requiring that the recovery process defines the functions (for example, by loading a q script) prior to replaying the log. 
The advantages can outweigh the disadvantages, however, by allowing for bugs fixes within the function, or temporarily assigning the function to a different definition prior to playback, to provide bespoke logic for data sourced from a log file.
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+
+A kdb+ process can run 24/7, but a log file may only be relevant for a specific timeframe or event. 
+For example, the default tickerplant creates a new log for each day. 
+The developer should close the current log and create a new one on each event. 
```
```suggestion
You should ensure the current log is closed and a new log created on each event. 
```

natalietanner commented on docs/kb/logging.md:
```diff
@@ -183,14 +185,88 @@ the replicating process will receive this information when it connects.
+
+A naming convention should be used to aid distinction between current log files and any old log files required for retention.
+The [`z`](../ref/dotz.md) namespace provides various functions for system information such as current date, time, etc. 
+The following is an example of namimg a log file after the current date:
```
```suggestion
The below example demonstrates naming a log file after the current date:
```

natalietanner commented on docs/kb/logging.md:
````diff
@@ -293,6 +369,35 @@ q)-11!(26;logfile)
+hclose goodfilehandle;
+```
+
+Alternatively, generic system tools can be used e.g. unix `head` command. For example, given that ``-11!(-2;`:bad.log)`` returns 2879 bytes.
````
```suggestion
Alternatively, generic system tools can be used like the unix `head` command. For example, given that ``-11!(-2;`:bad.log)`` returns 2879 bytes.
```

sshanks-kx committed:
Update docs/architecture/index.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/basics/ipc.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/basics/syscmds.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/basics/ipc.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update docs/kb/logging.md

Co-authored-by: natalietanner <142605282+natalietanner@users.noreply.github.com>

sshanks-kx committed:
Update logging.md

sshanks-kx committed:
Update logging.md

fix formatting, the note box was causing the numbered list to render incorrectly

sshanks-kx commented:
changes made

natalietanner commented:
Changes approved.
